### PR TITLE
Fix floating point exceptions in sgame

### PR DESCRIPTION
**Pull request by slipher**
Fix "invalid" and "zero divide" floating point exceptions in sgame.

### CODE REVIEW

File: src/sgame/sg_bot_nav.cpp [hunk 1]
```diff
@@ -569,8 +569,16 @@ static bool BotAvoidObstacles( gentity_t *self, glm::vec3 &dir )
 	return true;
 }
 
+// `dir` does not need to be normalized
 static void BotDirectionToUsercmd( gentity_t *self, const glm::vec3 &dir, usercmd_t *cmd )
 {
+	if ( glm::length2( glm::vec2(dir) ) < 1.0e-5f )
```
Comment (DolceTriade) [inline src/sgame/sg_bot_nav.cpp:575]: Don't we have some epsilon defined already? Did you pick this number arbitrarily?

**Review by DolceTriade — APPROVED** [review]

File: src/sgame/sg_bot_nav.cpp (hunk 1 above)
Comment (slipher) [reply src/sgame/sg_bot_nav.cpp:575]: It seems we have dozens of epsilons defined, though none particularly appropriate. In this case I think the epsilon could be many orders of magnitude bigger or smaller in either direction and not matter.

File: src/sgame/sg_bot_util.cpp [hunk 2]
```diff
@@ -1538,6 +1538,10 @@ void BotSlowAim( gentity_t *self, glm::vec3 &target, float slowAmount )
 	//get the Vector from the bot to the enemy (ideal aim Vector)
 	glm::vec3 aimVec = target - viewBase;
 	float length = glm::length( aimVec );
+	if ( length < 1.0f )
+	{
+		return;
+	}
```
Comment (necessarily-equal) [inline src/sgame/sg_bot_util.cpp:1544]: I'm genuinely wondering when this happens, but why not I guess

**Review by necessarily-equal — APPROVED** [review]
LGTM

File: src/sgame/sg_bot_util.cpp (hunk 2 above)
Comment (slipher) [reply src/sgame/sg_bot_util.cpp:1544]: It does happen. If you want to know, set a breakpoint there :stuck_out_tongue:

**Pull request merged**